cmd/podman/common/completion.go

```diff
@@ -1183,6 +1183,38 @@ func AutocompleteFormat(o interface{}) func(cmd *cobra.Command, args []string, t
 			return nil, cobra.ShellCompDirectiveNoFileComp
 		}
 
+		// add suggestions for: podman inspect (it will default to container inspect)
+		switch reflect.TypeOf(o).String() {
+		case "*define.InspectContainerData":
```

Comment:
I don't think this can work, this type depends on what you pass in `AutocompleteFormat()`. If we want to support more than container than you need to have more than one type and you cannot know that until this function is called.

You need to basically invert the order first check getContainers() if the arg is a container then you can set
`o = &define.InspectContainerData{}` same for the other types and so on.

Because AutocompleteFormat() is used by a lot of different callers we need to have a check to only do that for the podman inspect command. One way to do that without adding a new parameter is to check for
`cmd.Name()` == `inspect` and `cmd.Parent() == cmd.Root()`. Only then we want to check the arg and overwrite the `o` value.